lib/chef/knife/windows_cert_generate.rb

```diff
@@ -75,7 +76,7 @@ def prompt_for_passphrase
       end
 
       def generate_certificate rsa_key
-        @hostname = "*"
+        @hostname = Socket.gethostname
```

Comment:
I'm feeling like we should make it mandatory and not have a default. The user knows what the fqdn is that they're going to try to connect to right after they make this certificate, which is our best chance to be right most of the time with the least trouble.

I just looked at what [ohai does to get fqdn on windows](https://github.com/chef/ohai/blob/master/lib/ohai/plugins/hostname.rb#L163) and on a couple of my systems it just returns the hostname, even for fqdn. Doing a reverse DNS lookup requires that the user 1) has DNS correctly set up and 2) that the reverse DNS matches what it should. It's really common for reverse DNS to return nothing (or a generalized name based on ip address), in which case we would want to be sure to fail early. The hostname that the user is connecting to to get to the server isn't necessarily the reverse DNS either, e.g. reverse dns returns unique-asset-id.example.org, but I'm connecting to service-name.example.org.